.github/workflows/main.yml

```diff
@@ -19,6 +19,11 @@ jobs:
         uses: actions/setup-python@v2
         with:
           python-version: "^3.9"
+      - name: Load cached Poetry installation
+        uses: actions/cache@v2
+        with:
+          path: ~/.local  # the path depends on the OS
+          key: dotlocal-${{ runner.os }}-${{ hashFiles('.github/workflows/main.yml') }}
```

Comment:
Should this check for a hash of `pyproject.toml` instead of `main.yml`? I.e., wouldn't the Poetry version be defined in the `pyproject.toml`? 

I'm just not sure how this Poetry caching step should work.

Reply:
Nevermind, I see that the specific Poetry version is defined later in this file.